to.etc.domui/src/main/java/to/etc/domui/dom/HtmlTagRenderer.java

```diff
@@ -677,9 +677,11 @@ public void basicNodeRender(final NodeBase b, final IBrowserOutput o, boolean in
 			UIDragDropUtil.exposeDroppable(b, dh);
 		}
 
-		String s = getStyleFor(b); 								// Get/recalculate style
-		if(s.length() > 0)
-			o.attr("style", s); 								// Append style
+		String cachedStyles = b.getCachedStyle();
```

Comment:
Well, the logic here is very hard to follow, so this will break easily....

I think it is still wrong. When a completely new tree is rendered then cachedStyle is also null. In that case we render a style="" on every node that has no attributes.

I might have a solution to this but it's better to talk through it. Call me when you find the time (unless you can convince me that this code is correct, of course ;)